addons/udes_security/models/res_archiving_restrictions.py

```diff
@@ -0,0 +1,82 @@
+from odoo import fields, models, _, SUPERUSER_ID
+from odoo.exceptions import ValidationError
+
+
+class ResArchivingRestrictions(models.Model):
+    """
+    Configurable access to _only_ record archiving & unarchiving
+
+    If an Archiving Restriction is created for a model, the group_ids set will determine
+    which groups have the facility to archive and unarchive records on the model.
+
+    This includes backend writes to the `active` field, as well as hiding
+    the Archive/Unarchive action from the model, and attempting to remove active
+    fields & smart buttons from form views.
+    """
+
+    _name = "res.archiving.restrictions"
```

Comment:
Usually the names of records are singular, e.g. `res.archiving.restriction`.

I know Odoo is inconsistent with this itself, but I think it should still be singular here, as we have one record per model that is restricted.

Reply:
Yeah, i wondered about this because even I was being inconsistent with naming and `res.archiving.restriction` seemed slightly better, but didn't want to waste time doing it if it was going to be scrapped anyway. Will get it changed (Y)